src/http/authentication/cert_auth.h

```diff
@@ -34,21 +32,17 @@ namespace ccf
       const auto& caller_cert = ctx->session->caller_cert;
       auto caller_id = crypto::Sha256Hash(caller_cert).hex_str();
 
-      auto users = tx.ro<Users>(Tables::USERS);
-      const auto user = users->get(caller_id);
-      if (user.has_value())
+      auto user_certs = tx.ro<UserCerts>(Tables::USER_CERTS);
+      const auto user_cert = user_certs->get(caller_id);
```

Comment:
If we're retrieving (and deserialising) the user data lazily, it seems like we could also get a slight perf win (/consistency gain) from making the user cert lazy as well? Or in fact, we could use the `caller_cert` here, and not deserialise from the `UserCerts` table, since they "must" match!

Reply:
I was wondering about that. My original hunch was it's nicer to always present the PEM cert to users, rather than the DER-encoded `caller_cert`. The `UserCerts` lookup is basically a cheap (*) way to encode the caller certificate as PEM. I'm not sure there are any advantages to have the caller cert as PEM rather than DER anymore though, so I'm keen to simplify this too.

(*) I've checked and a pure DER to PEM conversion (crypto::cert_der_to_pem(der)`) costs ~2/3 of performance on the smallbank benchmark.

Reply:
So I think this is an extra argument for making this lazy. We're not sure if we should be presenting the cert as DER or PEM, because we don't know what an app might want to use, and in fact we have no apps which use either. I think we should stop trying to present one 'for free' (because we've seen that even this deserialisation from the `UserCerts` lookup is not free), and instead have `get_cert_for_user_v1` (or perhaps more precisely `get_pem_for_user_v1` and/or `get_der_for_user_v1`, but I think `get_cert` is fine and `crypto::` is part of our supported API).

Reply:
Done. Added two new `get_user_cert_v1()` and `get_member_cert_v1()` API calls.